client/src/utils/tone/index.ts

```diff
@@ -0,0 +1,36 @@
+import store from 'store';
+import { FlashMessages } from '../../components/Flash/redux';
+import { Themes } from '../../components/settings/theme';
+
+const toneUrls = {
+  [FlashMessages.IncompleteSteps]:
+    'https://campfire-mode.freecodecamp.org/try-again.mp3',
+  [FlashMessages.CertClaimSuccess]:
+    'https://campfire-mode.freecodecamp.org/cert.mp3',
+  [Themes.Default]: 'https://campfire-mode.freecodecamp.org/day.mp3',
+  [Themes.Night]: 'https://campfire-mode.freecodecamp.org/night.mp3',
+  donation: 'https://campfire-mode.freecodecamp.org/donate.mp3',
+  'tests-completed': 'https://campfire-mode.freecodecamp.org/chal-comp.mp3',
+  'block-toggle': 'https://tonejs.github.io/audio/berklee/guitar_chord1.mp3',
+  'tests-failed': 'https://campfire-mode.freecodecamp.org/try-again.mp3',
+  completion: 'https://campfire-mode.freecodecamp.org/chal-comp.mp3'
+} as const;
+
+type ToneStates = keyof typeof toneUrls;
+
+export async function playTone(state: ToneStates): Promise<void> {
+  const playSound = !!store.get('fcc-sound');
+  if (playSound) {
+    const tone = await import('tone');
+    if (tone.context.state !== 'running') {
+      tone.context.resume().catch(err => {
+        console.error('Error resuming audio context', err);
+      });
+    }
+
+    if (toneUrls[state]) {
+      const player = new tone.Player(toneUrls[state]).toDestination();
+      player.autostart = true;
+    }
+  }
```

Comment:
```suggestion
  if (playSound && toneUrls[state]) {
    const tone = await import('tone');
    if (tone.context.state !== 'running') {
      tone.context.resume().catch(err => {
        console.error('Error resuming audio context', err);
      });
    }
    const player = new tone.Player(toneUrls[state]).toDestination();
    player.autostart = true;
  }
```

Can we move the URL validation into the first condition? This way we're only importing + loading Tone if the camper has the mode enabled AND it's a valid URL to play.

Reply:
I suppose the question is, when is `toneUrls[state]` false?

Reply:
When the flash message gets called for something we don't have a sound for, is what comes to mind initially.